lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -657,8 +657,12 @@ sub overview {
         groups => $groups,
         until => $until,
     );
-    my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
-    ($stash{archs}, $stash{results}, $stash{aggregated}) = $self->prepare_job_results(\@latest_jobs);
+    my $limit = $search_args->{limit} = OpenQA::App->singleton->config->{misc_limits}->{tests_overview_max_jobs};
+    my $jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args);
+    my $limit_exceeded = $jobs->count >= $limit ? $limit : undef;
+    my @latest_jobs = $jobs->latest_jobs($until);
```

Comment:
The limit might not be exceeded if `$until` is given, but that is only used after doing the job count and check for limit.

Reply:
I've tried that actually, the limit was much less reliable then and did not work for the case the ticket is about.

Reply:
The problem is the two resultsets being used here, one created by `complex_query` and the other by `latest_jobs`. Without a limit i get 82991 for the first and 67539 for the second. With a limit on the first query i get 500 for the first and 468 for the second. If i apply the limit to the second only i get 82991 for the first and 468 for the second. 😁 Good luck trying to detect when the limit clause actually triggers!

Reply:
Changed the API a bit to make it work.